lsy_drone_racing/utils/checks.py

```diff
@@ -16,66 +15,125 @@
+
+    low, high = rng_config.gate_pos.kwargs.minval, rng_config.gate_pos.kwargs.maxval
+    for i, (pos, nominal_pos) in enumerate(zip(gates.pos, gates.nominal_pos)):
+        check_bounds(f"gate{i + 1}", pos, nominal_pos, np.array(low), np.array(high))
+
+    # TODO: Now the gate check should consider rotation in roll and pitch as well.
+    high_tol = np.array(rng_config.gate_rpy.kwargs.maxval)
+    low_tol = np.array(rng_config.gate_rpy.kwargs.minval)
+    # ang_tol = rng_config.gate_rpy.kwargs.maxval[2]  # Only check yaw rotation
+    for i, (quat, nominal_quat) in enumerate(zip(gates.quat, gates.nominal_quat)):
+        gate_rot = R.from_quat(quat)
+        nominal_rot = R.from_quat(nominal_quat)
+        check_rotation(f"gate{i + 1}", gate_rot, nominal_rot, low=low_tol, high=high_tol)
+
+    low, high = rng_config.obstacle_pos.kwargs.minval, rng_config.obstacle_pos.kwargs.maxval
+    for i, (pos, nominal_pos) in enumerate(zip(obstacles.pos, obstacles.nominal_pos)):
+        check_bounds(
+            f"obstacle{i + 1}", pos[:2], nominal_pos[:2], np.array(low[:2]), np.array(high[:2])
+        )
+
+
+def check_drone_start_pos(
+    nominal_pos: NDArray, real_pos: NDArray, rng_config: ConfigDict, drone_name: str
+):
     """Check if the real drone start position matches the settings.
 
     Args:
-        pos: Current drone position.
+        nominal_pos: Nominal drone position.
+        real_pos: Current drone position.
         rng_config: Environment randomization config.
         drone_name: Name of the drone (e.g. cf10).
     """
     assert rng_config.drone_pos.fn == "uniform", (
         "Drone start position check expects uniform distributions"
     )
     tol_min, tol_max = rng_config.drone_pos.kwargs.minval, rng_config.drone_pos.kwargs.maxval
-    ros_connector = ROSConnector(estimator_names=[drone_name], timeout=5.0)
-    try:
-        real_pos = ros_connector.pos[drone_name]
-    finally:
-        ros_connector.close()
-    check_bounds(drone_name, real_pos[:2], pos[:2], tol_min[:2], tol_max[:2])
+
+    check_bounds(
+        drone_name, real_pos[:2], nominal_pos[:2], np.array(tol_min[:2]), np.array(tol_max[:2])
+    )
 
 
 def check_bounds(name: str, actual: NDArray, desired: NDArray, low: NDArray, high: NDArray):
-    """Check if the actual value is within the specified bounds of the desired value."""
+    """Check if the actual value is within the specified bounds of the desired value.
+
+    Raise error when the actual value exceeds the low or high boundaries.
+
+    Args:
+        name: Name of the object being checked.
+        actual: NDArray of real value and is expected to be [N, ].
+        desired: NDArray of nominal value and is expected to be [N, ].
+        low: NDArray of the minimum value of (actual - desired) and is expected to be [N, ]
+        high: NDArray of the maximum value of (actual - desired) and is expected to be [N, ]
+    """
```

Comment:
Nothing in this function assumes a shape. Also, do not repeat types in the doc string. They are contained in the function signature already.

```suggestion
    """Check if the actual value is within the specified bounds of the desired value.

    Args:
        name: Name of the object being checked.
        actual: Values to check.
        desired: Reference values.
        low: Lower bound. Minimum permissible value of (actual - desired).
        high: Upper bound. Maximum value of (actual - desired).
    
    Raises:
            RuntimeError: The values are not in the permissible interval.
    """
```